packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -178,68 +45,24 @@ function getWrappedShadowRootListener(
     sr: SyntheticShadowRootInterface,
     listener: EventListener
 ): WrappedListener {
-    if (!isFunction(listener)) {
-        throw new TypeError(); // avoiding problems with non-valid listeners
-    }
     let shadowRootWrappedListener = shadowRootEventListenerMap.get(listener);
     if (isUndefined(shadowRootWrappedListener)) {
         shadowRootWrappedListener = function(event: Event) {
-            // * if the event is dispatched directly on the host, it is not observable from root
-            // * if the event is dispatched in an element that does not belongs to the shadow and it is not composed,
-            //   it is not observable from the root
-            const { composed } = event as any;
+            // if the event is dispatched directly on the host, it is not observable from root,
+            // this is a limitation of the synthetic shadow polyfill
```

Comment:
We briefly discussed offline that this should read more like, "if the event is dispatched directly on the shadow root, it is not observable from the root", but I don't think we support dispatching events on shadow roots, do we?

Reply:
thinking more about this... I think we can fix it by patching dispatchEvent on the shadowRoot, in which case, we know...